lots of comments added. not all though, as I didn't understand everyt…

kartoffelmozart opened the pull request:
…hing.

kartoffelmozart committed:
lots of comments added. not all though, as I didn't understand everything.

palleBrandt reviewed (approved):
Looks good. Very nice with all these comments

RazzerTazzer reviewed (approved):
I think this is a perfect amount of comments in the code. Fun to see the changes to followersfollower again. I like it

allanzigge commented on src/Chirp.Infrastructure/AuthorValidation.cs:
```diff
@@ -2,8 +2,10 @@ namespace Chirp.Infrastructure;
+        // Set up the rules that a valid Author entity must follow.
         RuleFor(author => author.AuthorId).NotNull();
         RuleFor(author => author.Name).NotEmpty();
         RuleFor(author => author.Email).NotEmpty();
```
Not relevant for this issue, but I though if email check should be an email check, and not just NotEmpty?

kartoffelmozart replied:
Totalt enig, skal jeg lige ændre det og comitte? Eller skal den i sit eget issue først?

allanzigge replied:
Googlede hurtigt og fluent validation har vist en funktion der tjekker om den inderholder @, så ja og nej @Maquidos

palleBrandt replied:
Man kan gøre noget ala:

RuleFor(customer => customer.Email).EmailAddress();

allanzigge reviewed (changes requested):
Mega fedt med en masse kommentarer, jeg har foreslået enkelte ændringer :)

kartoffelmozart committed:
Update src/Chirp.Infrastructure/CheepRepository.cs, suggested by Allan

Co-authored-by: Allan: <asia@itu.dk>

Co-authored-by: Allan Sigge <118464722+allanzigge@users.noreply.github.com>

kartoffelmozart committed:
Update src/Chirp.Infrastructure/CheepRepository.cs change suggested by Allan

Co-authored-by: Allan: <asia@itu.dk>

Co-authored-by: Allan Sigge <118464722+allanzigge@users.noreply.github.com>

kartoffelmozart committed:
Update src/Chirp.Infrastructure/CheepRepository.cs suggested by Allan

Co-authored-by: Allan: <asia@itu.dk>

Co-authored-by: Allan Sigge <118464722+allanzigge@users.noreply.github.com>

kartoffelmozart committed:
Update src/Chirp.Infrastructure/CheepRepository.cs change suggested by Allan

Jeg tænkte "so no invalid data will reach the database" men din løsning er god
Co-authored-by: Allan: <asia@itu.dk>

Co-authored-by: Allan Sigge <118464722+allanzigge@users.noreply.github.com>

kartoffelmozart commented on src/Chirp.Infrastructure/CheepRepository.cs:
```diff
@@ -81,26 +100,34 @@ public async Task CreateCheep(string message, AuthorDto user)
     {
+        // Return a CheepDto of the first cheep from the given author.
+
+        // The first cheep of the given author.
```
Yeah kinda redundant, men det følger en konvention jeg fik afvide på et tidspunkt at lige under signaturen på hver metode skal der stå hvad metoden gør. Jeg ved ikke om det er good practice tho.

kartoffelmozart committed:
Update src/Chirp.Web/Pages/Discover.cshtml.cs change suggested by Allan

Co-authored-by: Allan: <asia@itu.dk>

Co-authored-by: Allan Sigge <118464722+allanzigge@users.noreply.github.com>

kartoffelmozart committed:
Update src/Chirp.Infrastructure/CheepRepository.cs change suggested by Allan

Co-authored-by: Allan: <asia@itu.dk>

Co-authored-by: Allan Sigge <118464722+allanzigge@users.noreply.github.com>

kartoffelmozart committed:
Update src/Chirp.Infrastructure/CheepRepository.cs suggested by Markus

Co-authored-by: Markus: <mbrh@itu.dk>

Co-authored-by: Markus Brandt Højgaard <118807770+Maquidos@users.noreply.github.com>

kartoffelmozart committed:
Update src/Chirp.Infrastructure/CheepRepository.cs suggested by Markus

Co-authored-by: Markus: <mbrh@itu.dk>

Co-authored-by: Markus Brandt Højgaard <118807770+Maquidos@users.noreply.github.com>

allanzigge commented on src/Chirp.Infrastructure/CheepRepository.cs:
```diff
@@ -59,7 +59,9 @@ public async Task<List<CheepDto>> GetCheepsFromAuthor(int page, string author)
     public async Task<List<CheepDto>> GetCheepsUserTimeline(int page, string UserName, List<Guid> authorIds)
     {
-        // Return cheeps from a given author and cheeps from authors that is followed by the user on specific page
+        // If it is not the users timeline, when the authorIds list is empty the method only returns the cheep of the Author.
```
Sorry jeg forstår altså ikke den her

Vi kan måske se på det i morgen

RazzerTazzer committed:
Hello my fix

Maquidos reviewed (approved):
looks good
